week03/assignment03_test.go

```diff
@@ -0,0 +1,357 @@
+package week03
+
+import (
+	"testing"
+)
+
+func Test_Rate_1(t *testing.T) {
+	t.Parallel()
+
+	m1 := Movie{
+		Name:   "Avengers",
+		Length: 185,
+	}
+
+	var rating float32 = 8.2
+	act := m1.Rate(rating)
+
+	exp := "can't review a movie without watching it first"
```

Comment:
Using error messages as assertions is non-idiomatic in Go. Error messages change all the time and may contain dynamic information. Since we haven't covered errors in full yet, it would be enough to assert that an error does/does not exist.

```
if err == nil {
  t.Fatal("expected an error and got none")
}
```